### PR TITLE
GitHub use case

### PR DESCRIPTION
**Pull request by LoryApiFortress**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/api-testing/composer/io-components.md [hunk 1]
```diff
@@ -262,7 +263,7 @@ The `Contract Test` component has two child: the `Include (Contract T.)` and the
 
 ### File DataSource
 
-The **File Data Source** component allows you to use a file from the [Vault](/api-testing/vault/) as a data source.
+The **File Data Source** component allows you to use a file from the [Vault](/api-testing/vault/) as a data source. To [learn more](/api-testing/use-cases/use-drive) check the use case out.
```
Comment (LoryApiFortress) [inline docs/api-testing/composer/io-components.md:266]:
```suggestion
The **File Data Source** component allows you to use a file from the [Vault](api-testing/vault/#files) as a data source. To [learn more](/api-testing/use-cases/use-drive), check the use case out.
```

Comment (LoryApiFortress) [inline docs/api-testing/composer/io-components.md:266]:
```suggestion
The **File Data Source** component allows you to use a file from the [Vault](/api-testing/vault/#files) as a data source. To [learn more](/api-testing/use-cases/use-drive) check the use case out.
```

File: docs/api-testing/composer/io-components.md [hunk 2]
```diff
@@ -112,7 +113,7 @@ Performs a `DELETE` method request. See [Request Body](#request-body), [Request
 
 ### GitHub
 
-The GitHub component is meant to simplify the process of retrieving a file from GitHub and use it as a data source. Some examples of files to use would be CSV or JSON files. [Here is a tutorial](/api-testing/on-prem/how-to/github-for-datasets) on how to use it as part of a test.
+The GitHub component is meant to simplify the process of retrieving a file from GitHub and use it as a data source. Some examples of files to use would be CSV or JSON files. Check [Use Cases](/api-testing/use-cases/github-datasets/) out to see an example using this component.
```
Comment (LoryApiFortress) [inline docs/api-testing/composer/io-components.md:116]:
```suggestion
The GitHub component is meant to simplify the process of retrieving a file from GitHub and using it as a data source. Some examples of files to use would be CSV or JSON files. Check [Use Cases](/api-testing/use-cases/github-datasets/) out to see an example using this component.
```

File: sidebars.js [hunk 3]
```diff
@@ -1107,17 +1107,20 @@ module.exports = {
                         'api-testing/use-cases/key-value',
                         'api-testing/use-cases/integration-test',
                         'api-testing/use-cases/compose-body',
+                        'api-testing/use-cases/contract-test',
```
Comment (LoryApiFortress) [inline sidebars.js:1110]:
```suggestion
```

File: docs/api-testing/use-cases/github-datasets.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+When creating the token make sure you're selecting the **`public_repo`** subcategory or the **`repo`** category based on which type of repository that will host your input data.
+
+<img src={useBaseUrl('img/api-testing/io-components/type.png')} alt="Type of repository" width="400"/>
+:::
+
+2. Create a repository that will host your input data if one doesn't exist already.
+
+3. Commit and push a data source file. It can be a CSV, JSON or XML file. The example is based on the [bypopulation.csv](https://github.com/LoryApiFortress/examples/blob/7f03771e2d728da4546fcbc218b31642c5d9265a/bypopulation.csv) file.
+
+## Step 2: Creating a Simple Test
```
Comment (LoryApiFortress) [inline docs/api-testing/use-cases/github-datasets.md:37]:
```suggestion
## Step 2: Creating a Basic Test
```

File: docs/api-testing/use-cases/github-datasets.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+<img src={useBaseUrl('img/api-testing/io-components/github-report.png')} alt="test report" />
+
+5. Next step, iterate over a subset of this input set by adding an **Each**.
+
+<img src={useBaseUrl('img/api-testing/io-components/github-each.png')} alt="each component" />
+
+This will iterate over a subset of 5 randomly selected items.
+
+:::tip Suggestion
+To help you identify which item you are looking at for debugging purposes, you should add a comment that prints out the current item, within each iteration. (like `ID: ${_1[0]}`)
```
Comment (LoryApiFortress) [inline docs/api-testing/use-cases/github-datasets.md:70]:
```suggestion
To help you identify which item you are looking at for debugging purposes, you should add a comment that prints out the current item in each iteration. (like `ID: ${_1[0]}`)
```

**Comment by LoryApiFortress** [comment]
@aliciakrebs please review. Thanks

**Review by aliciakrebs — APPROVED** [review]
I committed some style and wording changes from my local. Please make sure that my changes  did not introduce any issues.  : ) Otherwise, this is good to merge.

File: docs/api-testing/use-cases/github-datasets.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+You can use CSV, JSON, or XML files that are stored in a GitHub repository as a data source in your tests. Follow this guide to learn how to use this [bypopulation.csv](https://github.com/LoryApiFortress/examples/blob/7f03771e2d728da4546fcbc218b31642c5d9265a/bypopulation.csv) file as a data source.
+
+## What You'll Need
+
+- A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up)).
+- An existing API Testing Project. For details on how to create one, see [API Testing Quickstart](/api-testing/quickstart/).
+
+## Step 1: Setting Up GitHub
+
+1. Create a new access token associated to your GitHub profile by accessing [https://github.com/settings/tokens](https://github.com/settings/tokens).
```
Comment (LoryApiFortress) [inline docs/api-testing/use-cases/github-datasets.md:19]:
```suggestion
1. Create a new access token associated to your GitHub profile by accessing [https://github.com/settings/tokens](https://github.com/settings/tokens). GitHub will show you the token only once, so make sure you’re copying and pasting it somewhere safe and accessible.
```

File: docs/api-testing/use-cases/github-datasets.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+<img src={useBaseUrl('img/api-testing/io-components/github-report.png')} alt="test report" />
+
+5. Next, iterate over a subset of this input set by adding an **Each** component.
+
+<img src={useBaseUrl('img/api-testing/io-components/github-each.png')} alt="each component" />
+
+This iterates over a subset of five randomly selected items.
+
+:::tip Suggestion
+To help you identify an item in a comment for debugging purposes, you can add a comment that prints out the current item in each iteration. (like `ID: ${_1[0]}`)
```
Comment (LoryApiFortress) [inline docs/api-testing/use-cases/github-datasets.md:66]:
```suggestion
To help you identify an item for debugging purposes, you can add a comment that prints out the current item in each iteration. (like `ID: ${_1[0]}`)
```

**Pull request merged**